x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx

```diff
@@ -49,7 +50,8 @@ const formatMetric = (type: SnapshotMetricInput['type'], value: number | undefin
 };
 
 const buildItemsList = (nodes: SnapshotNode[]) => {
-  return nodes.map(({ metrics, path, name }) => ({
+  return nodes.map(({ metrics, path, name }, index) => ({
```

Comment:
I got your point and I agree that in general having a unique key for list items is a good idea - I used the index here as it represents the order in the table - so it is fine to change the position and still have the same open tab: 

https://user-images.githubusercontent.com/14139027/226397891-312a64f4-6d6c-4858-bc4e-427b6e1407b4.mov


I will still take a look and try to generate unique ids instead of the index, I was afraid that adding complexity here may decrease the performance.

Reply:
I changed that to use uuidv4 and mocked the value in the test 🙂